x-pack/plugins/security_solution/server/lib/detection_engine/notifications/legacy_saved_object_references/legacy_extract_rule_id.test.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { loggingSystemMock } from 'src/core/server/mocks';
+// eslint-disable-next-line no-restricted-imports
+import { LegacyRulesNotificationParams } from '../legacy_types';
+// eslint-disable-next-line no-restricted-imports
+import { legacyExtractRuleId } from './legacy_extract_rule_id';
+
+describe('legacy_extract_rule_id', () => {
+  type FuncReturn = ReturnType<typeof legacyExtractRuleId>;
+  let logger = loggingSystemMock.create().get('security_solution');
+
+  beforeEach(() => {
+    logger = loggingSystemMock.create().get('security_solution');
+  });
+
+  test('it returns an empty array given a "undefined" ruleAlertId.', () => {
+    expect(
+      legacyExtractRuleId({
+        logger,
+        ruleAlertId: undefined as unknown as LegacyRulesNotificationParams['ruleAlertId'],
+      })
+    ).toEqual<FuncReturn>([]);
+  });
+
+  test('logs expect error message if given a "undefined" ruleAlertId.', () => {
+    expect(
+      legacyExtractRuleId({
+        logger,
+        ruleAlertId: null as unknown as LegacyRulesNotificationParams['ruleAlertId'],
+      })
+    ).toEqual<FuncReturn>([]);
+
+    expect(logger.error).toBeCalledWith(
+      'Security Solution notification (Legacy) system "ruleAlertId" is null or undefined when it never should be. ,This indicates potentially that saved object migrations did not run correctly. Returning empty reference'
```

Comment:
Is there any action we can give the user if they hit here?

Reply:
Not easily. Proactive advice is hard here as we don't even know which alert id this would be associated with to tell them to re-do their notifications. It's just a lost notification.

If we get a way to auto re-create/migration notifications through a REST API we could revisit this error possibly to tell them to run that but in reality that would just delete/drop this as it would not know which alert id this belongs to.

Reply:
I did fix the `. ,This` to be just `. This` as that's not what we want in the error message.